arrow-core-data/src/main/kotlin/arrow/core/Ior.kt

```diff
@@ -486,8 +904,247 @@ fun <A> A.leftIor(): Ior<A, Nothing> = Ior.Left(this)
+  traverse_(::identity)
+
+fun <A, B, C> Ior<A, Either<B, C>>.sequenceEither_(): Either<B, Unit> =
+  traverseEither_(::identity)
+
+fun <A, B, C> Ior<A, Validated<B, C>>.sequenceValidated_(): Validated<B, Unit> =
+  traverseValidated_(::identity)
+
+fun <A, B> Ior<Iterable<A>, Iterable<B>>.bisequence(): List<Ior<A, B>> =
+  bitraverse(::identity, ::identity)
+
+fun <A, B, C> Ior<Either<A, B>, Either<A, C>>.bisequenceEither(): Either<A, Ior<B, C>> =
+  bitraverseEither(::identity, ::identity)
+
+fun <A, B, C> Ior<Validated<A, B>, Validated<A, C>>.bisequenceValidated(SA: Semigroup<A>): Validated<A, Ior<B, C>> =
+  bitraverseValidated(SA, ::identity, ::identity)
+
+fun <A, B> Ior<A, B>.combine(SA: Semigroup<A>, SB: Semigroup<B>, other: Ior<A, B>): Ior<A, B> =
+  with(SA) {
+    with(SB) {
+      when (val a = this@combine) {
+        is Ior.Left -> when (other) {
+          is Ior.Left -> Ior.Left(a.value + other.value)
+          is Ior.Right -> Ior.Both(a.value, other.value)
+          is Ior.Both -> Ior.Both(a.value + other.leftValue, other.rightValue)
+        }
+        is Ior.Right -> when (other) {
+          is Ior.Left -> Ior.Both(other.value, a.value)
+          is Ior.Right -> Ior.Right(a.value + other.value)
+          is Ior.Both -> Ior.Both(other.leftValue, a.value + other.rightValue)
+        }
+        is Ior.Both -> when (other) {
+          is Ior.Left -> Ior.Both(a.leftValue + other.value, a.rightValue)
+          is Ior.Right -> Ior.Both(a.leftValue, a.rightValue + other.value)
+          is Ior.Both -> Ior.Both(a.leftValue + other.leftValue, a.rightValue + other.rightValue)
+        }
+      }
+    }
+  }
+
+fun <A, B> Ior<A, B>.compare(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Ordering = fold(
+  { a1 -> b.fold({ a2 -> OA.run { a1.compare(a2) } }, { LT }, { _, _ -> LT }) },
+  { b1 -> b.fold({ GT }, { b2 -> OB.run { b1.compare(b2) } }, { _, _ -> LT }) },
+  { a1, b1 -> b.fold({ GT }, { GT }, { a2, b2 -> OA.run { a1.compare(a2) } + OB.run { b1.compare(b2) } }) }
+)
+
+fun <A, B> Ior<A, B>.compareTo(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Int =
+  compare(OA, OB, b).toInt()
+
+fun <A, B> Ior<A, B>.lt(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Boolean =
+  compare(OA, OB, b) == LT
+
+fun <A, B> Ior<A, B>.lte(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Boolean =
+  compare(OA, OB, b) != GT
+
+fun <A, B> Ior<A, B>.gt(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Boolean =
+  compare(OA, OB, b) == GT
+
+fun <A, B> Ior<A, B>.gte(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Boolean =
+  compare(OA, OB, b) != LT
+
+fun <A, B> Ior<A, B>.max(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Ior<A, B> =
+  if (gt(OA, OB, b)) this else b
+
+fun <A, B> Ior<A, B>.min(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Ior<A, B> =
+  if (lt(OA, OB, b)) this else b
+
+fun <A, B> Ior<A, B>.sort(OA: Order<A>, OB: Order<B>, b: Ior<A, B>): Pair<Ior<A, B>, Ior<A, B>> =
+  if (gte(OA, OB, b)) this to b else b to this
```

Comment:
I think we don't want to add all these methods. Might be similar for `Show`, `eqv` & `hash`.

Reply:
You're right, I will remove those methods and the instances of `Show`, `Eq`, `Order` and `Hash` for `Ior` datatype. In that case, are we just deprecating those typeclasses with a message saying they will be removed in version 0.13?

Reply:
Yes. I'm not sure if we need to elaborate anywhere why we're doing this for futher reference. Almost feels like the 0.13.0 release needs to be accompanied by a couple of blogposts to explain our rationale about Arrow's progression 😅